xonsh/base_shell.py

```diff
@@ -15,11 +13,10 @@
 
 
 class TeeOut(object):
-    """Tees stdout into the original sys.stdout and another buffer instance that is 
-    provided.
-    """
+    """Tees stdout into the original sys.stdout and another buffer."""
 
     def __init__(self, buf, *args, **kwargs):
+        _, _ = args, kwargs
```

Comment:
> Within Google the idiom is to assign unused args to _ to tell both lint and readers of the code that the args are deliberately being ignored.

Yea, this idiom is pretty common. And I understand what the linter is trying to do, but I think that it is a bit excessive here.

> Having said that I personally think the *args, **kwargs presence in this method are problematic. If the interface that TeeOut is expected to implement changes then the signature of its methods should change. Silently discarding unexpected positional and keyword args is a landmine waiting to hurt someone.

This is meant to duck type file handles like you might find in `io`. These have a lot of arguments, most of which won't be relevant. That is why I think capturing unused arguments is a valid strategy here. Unfortunately, you can't have two arguments with the same name and so doing `*_, **_` is not allowed.